### PR TITLE
[Fix] ErrorStorage agent calls

### PR DESCRIPTION
**Pull request by joaquinco**
- Improve `ErrorStorage.get_error_stats/1` by just fetching the error entry instead of the whole state and then getting the error entry.
- Change reset logic to `ErrorStorage.reset_stats/1` which resets an ErrorStorage entry and returns the previous occurrence.

### CODE REVIEW

File: lib/boom_notifier/error_storage.ex [hunk 1]
```diff
@@ -81,53 +75,59 @@ defmodule BoomNotifier.ErrorStorage do
...
   @doc """
   Reset the accumulated_occurrences for the given error info to zero. It also
   increments the max storage capacity based on the notification strategy.
+
+  Returns error storage entry before reset
   """
-  @spec reset_accumulated_errors(error_strategy, ErrorInfo.t()) :: :ok
-  def reset_accumulated_errors(:exponential, error_info) do
-    %{key: error_hash_key} = error_info
+  @spec reset(ErrorInfo.t()) :: __MODULE__.t()
```
Comment (iaguirre88) [inline lib/boom_notifier/error_storage.ex:89]: Loved the rename of this function 😍

File: lib/boom_notifier/error_storage.ex [hunk 2]
```diff
@@ -81,53 +75,59 @@ defmodule BoomNotifier.ErrorStorage do
...
+  end
 
-    Agent.update(
-      :boom_notifier,
-      &Map.update!(&1, error_hash_key, fn error_storage_item ->
-        clear_values(error_storage_item)
-        |> Map.update!(:__max_storage_capacity__, fn current -> min(current * 2, limit) end)
-      end)
-    )
+  def reset(error_info, strategy) when strategy in [:always, nil] do
```
Comment (iaguirre88) [inline lib/boom_notifier/error_storage.ex:101]: Maybe we can remove this guard so it always fallbacks here if the strategy doesn't need to update the `max_storage` value

File: lib/boom_notifier/error_storage.ex [hunk 3]
```diff
@@ -81,53 +75,59 @@ defmodule BoomNotifier.ErrorStorage do
...
+    reset_state(error_info, nil)
   end
 
-  def reset_accumulated_errors(:always, error_info) do
+  defp reset_state(error_info, nil) do
```
Comment (iaguirre88) [inline lib/boom_notifier/error_storage.ex:105]: NIT: `defp reset_state(error_info, _limit_updater_function = nil) do`

File: lib/boom_notifier/error_storage.ex [hunk 4]
```diff
@@ -37,19 +37,14 @@ defmodule BoomNotifier.ErrorStorage do
...
-
     Agent.update(
       :boom_notifier,
       &Map.update(
         &1,
         error_hash_key,
-        default_error_storage_info,
+        timestamp
+        |> default_error_storage()
+        |> Map.put(:accumulated_occurrences, 1),
```
[the remark this answers is not in the document]
Comment (joaquinco) [reply lib/boom_notifier/error_storage.ex:47]: good idea!

File: lib/boom_notifier/error_storage.ex (hunk 1 above)
Comment (germanbottosur) [reply lib/boom_notifier/error_storage.ex:92]: What about something like `ErrorStorage.reset_stats/2` or `ErrorStorage.reset_and_get_stats/2`

I'm thinking about the `notifications_sender` reading:

```elixir
occurrences = ErrorStorage.reset_stats(error_info, notification_trigger)
```

Comment (joaquinco) [reply lib/boom_notifier/error_storage.ex:92]: makes sense to be consistent with other ErrorStorage's functions

**Review by germanbottosur — APPROVED** [review]
Great work 🙌

**Comment by iaguirre88** [comment]
👏 👏 👏

**Pull request merged**